coremltools/converters/mil/mil/passes/defs/lower_complex_dialect_ops.py

```diff
@@ -338,52 +338,45 @@ def _stft(
             input_imaginary = mb.expand_dims(x=input_imaginary, axes=(0,), before_op=before_op)
 
     is_onesided = onesided and onesided.val
-    cos_base, sin_base = _calculate_dft_matrix(
-        n_fft,
-        onesided=is_onesided,
-        before_op=before_op)
+    cos_base, sin_base = _calculate_dft_matrix(n_fft, onesided=is_onesided, before_op=before_op)
     
     # create a window of centered 1s of the requested size
     if win_length:
-        n_left = (n_fft.val - win_length.val) // 2
-        n_right = n_fft.val - win_length.val - n_left
-
-        left = mb.fill(shape=(n_left,), value=0., before_op=before_op)
-        if not window:
-            window = mb.fill(shape=(win_length.val,), value=1., before_op=before_op)
-        right = mb.fill(shape=(n_right,), value=0., before_op=before_op)
-        
-        # concatenate
-        window = mb.concat(values=(left, window, right), axis=0, before_op=before_op)
+        window = _get_window(win_length=win_length, n_fft=n_fft, before_op=before_op)
 
     # apply time window
     if window:
         cos_base = mb.mul(x=window, y=cos_base, before_op=before_op)
         sin_base = mb.mul(x=window, y=sin_base, before_op=before_op)
 
-    # conv with DFT kernel across the input signal
-    sin_base = mb.sub(x=0., y=sin_base, before_op=before_op)
+
+    # Expand
     cos_base = mb.expand_dims(x=cos_base, axes=(1,), before_op=before_op)
     sin_base = mb.expand_dims(x=sin_base, axes=(1,), before_op=before_op)
     hop_size = mb.expand_dims(x=hop_length, axes=(0,), before_op=before_op)
-
     signal_real = mb.expand_dims(x=input_real, axes=(1,), before_op=before_op)
+    if input_imaginary:
+        signal_imaginary = mb.expand_dims(x=input_imaginary, axes=(1,), before_op=before_op)
+
+    # conv with DFT kernel across the input signal
+    # The DFT matrix is obtained with the equation e^(2pi/N i) but the definition is:
+    # DFT(x) => X[k] = Σx[n]*e^(-2kpi/N i)
+    # If x is complex then x[n]=(a+i*b)
+    # So the real part = Σ(a*cos(2kpi/N)+b*sin(2kpi/N))
+    # So the imag part = Σ(b*cos(2kpi/N)-a*sin(2kpi/N))
```

Comment:
Thank you for adding the comments here for better explanations! 💯

Reply:
I think now it's more legible